lib/elixir/test/elixir/integer_test.exs

```diff
@@ -12,9 +22,16 @@ defmodule IntegerTest do
     assert Integer.is_odd(-1) == true
     assert Integer.is_odd(-2) == false
     assert Integer.is_odd(-3) == true
+    assert test_is_odd_in_guards(10) == false
+    assert test_is_odd_in_guards(11) == 11
+    assert_raise ArgumentError,
+      "Integer.is_odd/1 expects an integer as an argument, got: 1.0", fn ->
+        n = fn -> 1.0 end
+        Integer.is_odd n.()
```

Comment:
This is a work around to avoid getting a message saying 
> warning: this expression will fail with ArithmeticError

I wonder if this is the right way to bypass it.